### ISSUE TITLE
Finer `one-var` control 

### ISSUE THREAD

**Issue by stevenvachon** [eslint/eslint#10234]
**What rule do you want to change?**
[`one-var`](https://eslint.org/docs/rules/one-var)

**Does this change cause the rule to produce more or fewer warnings?**
Either, depending on configuration.

**How will the change be implemented? (New option, new default behavior, etc.)?**
New values for `var`, `let` and `const`:
```json
"one-var": [2, {
    "const": "never",
    "let": {
        "initialized": "never",
        "uninitialized": "consecutive"
    }
}],
```

**Please provide some example code that this change will affect:**
```js
const a = 'value';
const b = 'value';
let c = 'value';
let d = 'value';
something(() => {
    c = 'new value';
    d = 'new value';
});

let e, f, g, h;
something(() => {
    e = 'new value';
    f = 'new value';
    g = 'new value';
    h = 'new value';
});
```

**What does the rule currently do for this code?**
Depending on the configuration, It either forces me to merge all definitions into a single declaration or into separate declarations, regardless of its initialized state.

**What will the rule do after it's changed?**
It will pass without error or warning.

**Comment by platinumazure** [comment]
Hi @stevenvachon, thanks for the issue. On behalf of the ESLint team, I apologize for letting this sit so long.

Could you please explain what problem you're trying to solve? It sounds like the `one-var` configuration is too limited in some way for your use case, but I don't fully understand what that is. Until I have a better understanding, I'm not sure I can evaluate the proposal.

**Comment by stevenvachon** [comment]
Thank you. It's not an issue now as I was able to achieve what I wanted with the current implementation using `{ initialized: 'never', uninitialized: 'consecutive' }`. `const a;` results in a parse error, so it doesn't present the problem I'd thought.
